src/lucky_cli/spinner.cr

```diff
@@ -0,0 +1,134 @@
+require "colorize"
+
+class LuckyCli::Spinner
+  DELETE_LINE = "\33[2K\r"
+
+  private property frame_index : Int32 = 0
+  private property? started : Bool = false
+  private property last_frame_rendered_at : Time? = nil
+
+  @done_text : String | ColoredString | Nil
```

Comment:
By default this deletes the line, but you can provide "done_text" to replace the spinner part with text: `Compiling... done`